utils/src/main/java/com/cloud/utils/nio/Link.java

```diff
@@ -615,7 +615,10 @@ public static boolean doHandshake(final SocketChannel socketChannel, final SSLEn
                 case NEED_TASK:
                     Runnable task;
                     while ((task = sslEngine.getDelegatedTask()) != null) {
-                        new Thread(task).run();
+                        if (s_logger.isTraceEnabled()) {
+                            s_logger.trace("SSL: Running delegated task!");
+                        }
+                        task.run();
```

Comment:
@jburwell thanks for pointing out, I've made the runnable task run in the current thread only. The delegated tasks in case of SSL handshake could be resolving domain names to address.